stacks/spica/src/passport/pages/api-key-add/api-key-add.component.spec.ts

```diff
@@ -0,0 +1,151 @@
+import {async, ComponentFixture, TestBed, tick, fakeAsync} from "@angular/core/testing";
+import {ApiKeyAddComponent} from "./api-key-add.component";
+import {
+  MatIconModule,
+  MatToolbarModule,
+  MatCardModule,
+  MatFormFieldModule,
+  MatInputModule
+} from "@angular/material";
+import {FormsModule} from "@angular/forms";
+import {NoopAnimationsModule} from "@angular/platform-browser/animations";
+import {ActivatedRoute, Router} from "@angular/router";
+import {of} from "rxjs";
+
+import {MockService} from "../../services/api-key.service";
+import {ApiKey} from "../../interfaces/api-key";
+
+describe("ApiKeyAddComponent", () => {
+  let component: ApiKeyAddComponent;
+  let fixture: ComponentFixture<ApiKeyAddComponent>;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        MatIconModule,
+        MatToolbarModule,
+        MatCardModule,
+        MatFormFieldModule,
+        MatInputModule,
+        FormsModule,
+        NoopAnimationsModule
+      ],
+      providers: [
+        {
+          provide: ActivatedRoute,
+          useValue: {
+            params: of({})
+          }
+        },
+        {
+          provide: Router,
+          useValue: {
+            navigate: () => {}
+          }
+        },
+        {
+          provide: MockService,
+          useValue: {
+            get: () => of(null),
+            update: () => {},
+            insert: () => {}
+          }
+        }
+      ],
+      declarations: [ApiKeyAddComponent]
+    }).compileComponents();
+  }));
+
+  beforeEach(async () => {
+    fixture = TestBed.createComponent(ApiKeyAddComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+    await fixture.whenStable();
+  });
+
+  it("should set apiKey as emptyApiKey if this page navigated from add button", () => {
+    expect(component.apiKey).toEqual({
+      name: undefined,
+      active: true
+    });
+  });
+
+  it("should set apiKey which return from service if this page navigated from edit button", async () => {
+    component["activatedRoute"].params = of({id: "1"});
+    component["apiKeyService"].get = () =>
```

Comment:
You can use spyOn here. 

Example;

```
spyOn(component['activatedRoute'], 'get').and.returnValue();
```

Reply:
I tried these ;
`spyOnProperty(component["activatedRoute"], "params", "get").and.returnValue(of({id: "1"}))`
`spyOn(component["activatedRoute"].params,"get").and.returnValue(of({id: "1"}))`
But the result was same **Property params does not have access type get**.
I didn't want to spend too much time for that.